### PR TITLE
Made ComboBox scrollable

### LINKED ISSUE #4727: combox box does not show all the options
```
ComboBox {
            model: [
                 10,  20,  30,  40,  50,  60,  70,  80,  90, 100,
                110, 120, 130, 140, 150, 160, 170, 180, 190, 200,
                210, 220, 230, 240, 250, 260, 270, 280, 290, 300,
                310, 320, 330, 340, 350, 360, 370, 380, 390, 400,
                410, 420, 430, 440, 450, 460, 470, 480, 490, 500,
                510, 520, 530, 540, 550, 560, 570, 580, 590, 600,
                610, 620, 630, 640, 650, 660, 670, 680, 690, 700,
                710, 720, 730, 740, 750, 760, 770, 780, 790, 800,
                810, 820, 830, 840, 850, 860, 870, 880, 890, 900,
                910, 920, 930, 940, 950, 960, 970, 980, 990, 1000
            ];
        }
```
I am able to get all the list of options available with comboBox. **how can i see all available options?**
Cargo.toml
```
[dependencies]
slint = "1.4"

[build-dependencies]
slint-build = "1.4"
```

### PR DESCRIPTION
**Pull request by ghost**
Fixes #4727

### CODE REVIEW

File: internal/compiler/widgets/qt/combobox.slint [hunk 1]
```diff
@@ -2,58 +2,64 @@
...
+    in property <[string]> model <=> base.model;
+    in property <bool> enabled <=> base.enabled;
+    out property <bool> has-focus <=> base.has-focus;
+    in-out property <int> current-index <=> base.current-index;
+    in-out property <string> current-value <=> base.current-value;
 
-    callback selected <=> i-base.selected;
+    callback selected <=> base.selected;
+    
+    property <length> item-height: 40px;
```
Comment (ogoffart) [inline internal/compiler/widgets/qt/combobox.slint:16]: Can this really be hardcoded like this? Where does the value comes from?

**Review by ogoffart — COMMENTED** [review]
I have concerned about the hardcoded size for the item height. Also only 5 visible items is really small.
Qt's comobbox has that configurable with the default being 10.

File: internal/compiler/widgets/qt/combobox.slint [hunk 2]
```diff
@@ -2,58 +2,64 @@
...
+        ListView {
+            width: parent.width;
+            height: parent.height;
 
             for value[index] in root.model: NativeStandardListViewItem {
                 item: { text: value };
                 is-selected: root.current-index == index;
                 has-hover: ta.has-hover;
                 combobox: true;
+                height: root.item-height;
```
Comment (ogoffart) [inline internal/compiler/widgets/qt/combobox.slint:58]: Even if we hardcode the size for the purpose of computing the maximum height, we shouldn't hardcode the size here.

**Review by ogoffart — COMMENTED** [review]
Yes indeed, we need to place the popup window differently if it is at the bottom of the screen.
This is complicated logic that needs support in the runtime and improvement to the PopupWindow which is unfortunately out of scope for the upcoming release.
I still think 5 is a bit small though, i'd prefer if it was a bit more.

And I'm not happy about the hardcoded size in the Qt style at all. You could get the size through NativeStyleMetrics perhaps. Anyway, i get your point that you don't like to touch this style too much.


Given that this is clearly an improvements for the other styles (as opposed to have the menu half clipped) i think we should merge this with a bit more elements visible and not set the `height` to the hardcoded size in the Qt style.

**Comment by ogoffart** [comment]
With https://github.com/slint-ui/slint/pull/5584 merged, it is now possible to position based on the size.

**Comment by ogoffart** [comment]
I think a hardcoded maximum size would be a good workaround for now. (without fixing the size of the `NativeStandardListViewItem`)

**Review by tronical — APPROVED** [review]

**Pull request merged**